src/standalone/Soil/energy_hydrology.jl

```diff
@@ -555,13 +556,31 @@ function ClimaLand.source!(
     (; ν, earth_param_set) = params
     _ρ_l = FT(LP.ρ_cloud_liq(earth_param_set))
     _ρ_i = FT(LP.ρ_cloud_ice(earth_param_set))
-    ρc = volumetric_heat_capacity.(p.soil.θ_l, Y.soil.θ_i, params)
-    τ = thermal_time.(ρc, src.Δz, p.soil.κ)
-
-    liquid_source =
-        phase_change_source.(p.soil.θ_l, Y.soil.θ_i, p.soil.T, τ, params)
-    @. dY.soil.ϑ_l += -liquid_source
-    @. dY.soil.θ_i += (_ρ_l / _ρ_i) * liquid_source
+    Δz_top = model.domain.fields.Δz_top # center face distance
```

Comment:
some rearrangement here to reduce allocations also - basically, putting it all on one line